src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -317,7 +316,7 @@ def load_arguments(self, _):
         c.argument('condition', is_preview=True, min_api='2020-04-01-preview', help='Condition under which the user can be granted permission.')
         c.argument('condition_version', is_preview=True, min_api='2020-04-01-preview', help='Version of the condition syntax. If --condition is specified without --condition-version, default to 2.0.')
         c.argument('assignment_name', name_arg_type,
-                   help='Name of the role assignment. If omitted, a new GUID is generetd.')
+                   help='The name of the role assignment. It must be unique and different for each role assignment. If omitted, a new GUID is generetd.')
```

Comment:
An interesting discovery:

That help message originally was

```
The name of the role assignment to create. It can be any valid GUID.
```

https://github.com/Azure/azure-rest-api-specs/pull/16639 updated it to 

```
A GUID for the role assignment to create. The name must be unique and different for each role assignment.
```

up to `2020-04-01-preview` API:

https://github.com/Azure/azure-rest-api-specs/blob/cf848c9f570c1157bbad81e214f5bb2993e578bb/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json#L242

```
          {
            "name": "roleAssignmentName",
            "in": "path",
            "required": true,
            "type": "string",
            "description": "A GUID for the role assignment to create. The name must be unique and different for each role assignment."
          },
```
but the change was reverted in `2022-04-01` API:

https://github.com/Azure/azure-rest-api-specs/blob/020f48c422ec455cdee9f6516f2d0172d13e77a3/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L693

```
    "RoleAssignmentNameParameter": {
      "name": "roleAssignmentName",
      "in": "path",
      "required": true,
      "type": "string",
      "description": "The name of the role assignment. It can be any valid GUID.",
      "x-ms-skip-url-encoding": true,
      "x-ms-parameter-location": "method"
    },
```

I am not sure why. `It can be any valid GUID.` is certainly not accurate.